### PR TITLE
feat(rome_js_analyze): add support for spread props in various rules

### PR DESCRIPTION
**Pull request by ematipico**
## Summary

This PR adds support for checking the presence of spread attributes inside some rules around JSX. The position where the spread attribute is positioned is important. These two snippets yield two different results.

```jsx
<div {...spread} onClick={}></div>
<div onClick={} {...spread} ></div>
```

Considering that we don't have any typing information around the spread props, it's safe to assume that when a spread attribute is **after** the attribute that triggers the rule, the rule should not be triggered anymore, because the spread attribute **might contain** the correct attribute.

## Test Plan

I added new test cases in the rules that have been affected

### CODE REVIEW

**Comment by xunilrj** [comment]
Should this be a rule on itself? "Spread should be at the beginning on JSX bla bla bla".

**Comment by ematipico** [comment]
> Should this be a rule on itself? "Spread should be at the beginning on JSX bla bla bla".

I think we had a rule that actually wanted to prevent the use of spread props, but it doesn't play well when using other components (like UI libraries) or library authors. If we had a some type information, we could make it a rule.

**Comment by ematipico** [comment]
@leops @xunilrj It would be great if you could give a look to this PR. This is needed in order to promote certain rules to be stable.

File: crates/rome_js_analyze/src/analyzers/nursery/use_blank_target.rs [hunk 1]
```diff
@@ -106,7 +124,9 @@ impl Rule for UseBlankTarget {
         if text.to_lowercase() == "_blank" {
             match rel_attribute {
                 None => {
-                    return Some((target_attribute, None));
+                    if !node.has_trailing_spread_attribute(target_attribute.clone())? {
```
Comment (MichaReiser) [inline crates/rome_js_analyze/src/analyzers/nursery/use_blank_target.rs:127]: Can we add a valid and invalid test case for this rule?

File: crates/rome_js_analyze/src/analyzers/nursery/use_key_with_click_events.rs [hunk 2]
```diff
@@ -18,6 +18,10 @@ declare_rule! {
     /// <div onClick={() => {}} ></div>
     /// ```
     ///
+    /// ```jsx,expect_diagnostic
+    /// <div {...spread} onClick={() => {}} ></div>
```
Comment (MichaReiser) [inline crates/rome_js_analyze/src/analyzers/nursery/use_key_with_click_events.rs:22]: Can we add a valid test case as well?

File: crates/rome_js_analyze/src/analyzers/nursery/use_valid_anchor.rs [hunk 3]
```diff
@@ -219,9 +235,15 @@ impl Rule for UseValidAnchor {
             (None, Some(on_click_attribute)) => Some(UseValidAnchorState::CantBeAnchor(
                 on_click_attribute.syntax().text_trimmed_range(),
             )),
-            (None, _) => Some(UseValidAnchorState::MissingHrefAttribute(
-                node.syntax().text_trimmed_range(),
-            )),
+            (None, None) => {
+                if !node.has_spread_attribute()? {
```
Comment (MichaReiser) [inline crates/rome_js_analyze/src/analyzers/nursery/use_valid_anchor.rs:239]: Nit
```suggestion
            (None, None) if !node.has_spread_attribute()? => {
```

File: crates/rome_js_analyze/src/analyzers/nursery/use_key_with_click_events.rs [hunk 4]
```diff
@@ -36,6 +40,10 @@ declare_rule! {
     /// // this rule doesn't apply to user created component
     /// <MyComponent onClick={() => {}} />
     ///```
+    ///
+    /// ```jsx,
+    /// <div onClick={() => {}} {...spread}></div>
```
[the remark this answers is not in the document]
Comment (ematipico) [reply crates/rome_js_analyze/src/analyzers/nursery/use_key_with_click_events.rs:45]: Yeah you're right, good catch

**Review by leops — APPROVED** [review]

**Pull request merged**